gov.usgs.warc.iridium.sbd.decoder/src/main/java/test/Echo.java

```diff
@@ -46,31 +54,40 @@ public static void main(final String... p_Args)
 			System.exit(1);
 		}
 
-		final String port = p_Args[0];
+		final String portS = p_Args[0];
+		final int port = Integer.parseInt(portS);
 		log.info(String.format("Listening on port %s", port));
-		try (final ServerSocket server = new ServerSocket(
-				Integer.parseInt(port));)
+		try (final ServerSocket server = new ServerSocket(port);)
 		{
 			while (true)
 			{
 				try (final Socket client = server.accept();)
 				{
 					log.info(String.format("Client connected: %s",
 							client.getInetAddress()));
-					final PrintWriter w = new PrintWriter(
-							client.getOutputStream(), true);
-					final String readLine = new BufferedReader(
-							new InputStreamReader(client.getInputStream()))
-									.readLine();
-					log.info(String.format("Message: %s", readLine));
-					w.println("ACK");
-					w.println(readLine);
+					try (OutputStream outputStream = client.getOutputStream();
+						OutputStreamWriter outputStreamWriter = new OutputStreamWriter(
+								outputStream, Charsets.UTF_8);
+						final PrintWriter w = new PrintWriter(
+								outputStreamWriter, true);
+						InputStream inputStream = client.getInputStream();
+						InputStreamReader inputStreamReader = new InputStreamReader(
+								inputStream, Charsets.UTF_8);
+						BufferedReader bufferedReader = new BufferedReader(
+								inputStreamReader);)
+					{
+						final String readLine = bufferedReader.readLine();
+						log.info(String.format("Message: %s",
+								readLine.replaceAll("\r\n", "\n")));
+						w.println("ACK");
+						w.println(readLine);
+					}
 				}
 			}
 		}
-		catch (final Exception e)
+		catch (final Throwable t)
 		{
-			log.error("Bad stuff happened.", e);
+			log.error("Bad stuff happened.", t);
```

Comment:
My favorite logging message in 9 years :)

Reply:
Ha!